Tools/CISupport/ews-build/steps.py

```diff
@@ -2585,12 +2586,25 @@ def analyzeResults(self):
                 self.build.buildFinished([message], FAILURE)
                 return defer.succeed(None)
 
-            message = 'Unable to build WebKit without {}, retrying build'.format('PR' if pr_number else 'patch')
+            retry_count = int(self.getProperty('retry_count', 0))
+            self.setProperty('retry_count', retry_count + 1)
+
+            message = 'Unable to build WebKit without {}, '.format('PR' if pr_number else 'patch')
+
+            if retry_count < CompileWebKitWithoutChange.MAX_RETRY:
+                message += 'retrying build'
+                result = RETRY
+            else:
+                message += 'please check manually'
+                result = FAILURE
+
             self.descriptionDone = message
             self.send_email_for_preexisting_build_failure()
-            self.finished(FAILURE)
```

Comment:
Before discarding the PR, does this bit, matching the `self.finished` and `self.build.buildFinished` results makes sense? (IIUC, this would match the behavior of other places where the build is retried, alongside the error messages)